app/controllers/needs_controller.rb

```diff
@@ -89,6 +110,31 @@ def request_destroy_note
     redirect_to need_path(note.need_id), notice: "You requested a manager to delete '#{note_name}'."
   end
 
+  def restore_need
+    need = policy_scope(Need).deleted.where(id: params[:id])
+    if need.exists?
+      need_name = need.first.category
+      Rails.logger.info("Restored need '#{params[:id]}'")
+      need.first.restore(:recursive => true)
+      redirect_to deleted_needs_path(order: "deleted_at", order_dir: "DESC"), 
+        notice: "Restored '#{need_name}' see <a href='/needs/#{need.first.id}'>here</a>"
```

Comment:
could use the URL helpers here, e.g. `#{need_path(need.first.id)}`

Reply:
And potentially:
`view_context.link_to` helper as well

Reply:
done